client/src/components/Header.js

```diff
@@ -275,16 +280,70 @@ export default class HeaderComponent extends Component {
             userInfo = await this.fetchUser()
             availableRoles = userInfo.result.role.map(role => {
                 role = role.toLowerCase()
+                let roleWords = role.split('_')
+                roleWords = roleWords.map(word => word[0].toUpperCase() + word.slice(1))
                 return {
                     key: role,
-                    text: role.charAt(0).toUpperCase() + role.slice(1),
+                    text: roleWords.join(' '),
                     value: role.toUpperCase()
                 }
             })
             this.setState({availableRoles: availableRoles})
         }
     }
 
+    welcomeBadge () {
+        switch (this.props.currentRole) {
+            case 'ALUMNI':
+                return (
+                    <Label
+                        style={{
+                            backgroundColor: black,
+                            color: 'white'
+                        }}
+                    >
+                        Welcome {this.props.name.split(' ')[0]} (Alumnus)
+                    </Label>
+                )
```

Comment:
This cases are all the same except for the name that it's given.. Seems like this could be its own small component where you pass in just the name you're trying to welcome and the role.
Looking at it more closely, if you pass the name + currentRole into this component, you can make this whole thing here just one line

Reply:
Nice catch! Changing

Reply:
Actually, I think there's among the 3 roles, with the string interpolation and Flag component, there are enough differences to warrant this explicit breakdown, what do you think?

Reply:
For the string interpolation, I think it's all the same except for wanting to put "Ambassador" in front of the country ambassador's name? You could check for that case pretty easily since you could pass in the role to the new component.

Also yeah, similar thing for the flag component, you could kinda do the same thing as above, but I'm wondering if there's a better way.. Is props.school.country defined for other types? If not, you could just pass that in and only show the flag if you passed it a country, or something.. Feel like this is kinda hard to work through over github, so happy to chat about it if you wanna discuss more